ui/app/src/state/epics/contentTypes.ts

```diff
@@ -65,21 +65,23 @@ export default [
     action$.pipe(
       ofType(fetchComponentsByContentType.type, setContentTypeFilter.type),
       withLatestFrom(state$),
-      switchMap(([, state]) =>
-        fetchItemsByContentType(
+      switchMap(([, state]) => {
+        const allowedContentTypes = Object.entries(state.preview.guest?.allowedContentTypes)
+          .filter(([, value]) => value.shared)
+          .map(([key]) => key);
+        return fetchItemsByContentType(
           state.sites.active,
-          state.preview.components.contentTypeFilter === 'all'
-            ? Object.values(state.contentTypes.byId)
-                .filter(
-                  (contentType: ContentType) =>
-                    contentType.type === 'component' && !contentType.id.includes('/level-descriptor')
-                )
-                .map((contentType) => contentType.id)
+          state.preview.components.contentTypeFilter === 'compatible'
+            ? allowedContentTypes?.length
+              ? allowedContentTypes
+              : // When there are no allowed content types, sending an empty array is like sending no content-type filter,
+                // so it returns items unfiltered. By sending an array with an empty string, we filter out all content-types.
+                ['']
```

Comment:
Maybe rather than this _hack_, the `fetchItemsByContentType` should not send the `content-type` filter if it's null or empty. This would allow having these checks removed all-across. Or ultimately, move this to the service if it makes sense; it is a bit of a yellow flag that this is in 2 places in the code.

Also, is this call exclusively to get content items by their content type? If all content types are filtered out, what is this returning? If this thing's job is to search by content type and we know in advanced the result is empty, then this shouldn't even be done and/or the service should have that smarts.